projects/views.py

```diff
@@ -1,3 +1,34 @@
-from django.shortcuts import render
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.permissions import IsAuthenticated
+from companies.permissions import IsCompanyMember
 
-# Create your views here.
+from projects.models import Project
+from projects.serializers import ProjectCreateUpdateSerializer, ProjectListSerializer
+
+class ProjectViewSet(ModelViewSet):
+    http_method_names = ('get', 'post', 'put', 'delete')
+    serializer_class = ProjectListSerializer
+    permission_classes = [IsAuthenticated]
+    queryset = Project.objects.all()
+    
+    def get_permissions(self):
+        match (self.action):
+            case 'update' | 'partial_update' | 'destroy':
+                return [IsAuthenticated(), IsCompanyMember()]
+            case _:       
+                return super().get_permissions()
+    
+    def get_queryset(self):
+        company_id = self.request.query_params.get("company_id")
+        if company_id:
+            return super().get_queryset().filter(company=company_id)
+
+        return super().get_queryset()
+
+    def get_serializer(self, *args, **kwargs):
+        match (self.action):
+            case 'create' | 'update' | 'partial_update':
+                kwargs['context'] = self.get_serializer_context()
+                return ProjectCreateUpdateSerializer(*args, **kwargs)
+            case _:
+                return super().get_serializer(*args, **kwargs)
```

Comment:
It's more idiomatic in DRF to override `get_serializer_class()` instead of `get_serializer()` unless you need dynamic context or manual instantiation:
```
def get_serializer_class(self):
    match self.action:
        case 'create' | 'update' | 'partial_update':
            return ProjectCreateUpdateSerializer
        case _:
            return ProjectListSerializer
```
This will work seamlessly with `.get_serializer()` later.